hail/python/hailtop/batch/job.py

```diff
@@ -452,25 +467,19 @@ def handler(match_obj):
                 _add_resource_to_set(self._valid, r)
 
             self._mentioned.add(r)
-            return f'{r_uid}'
-        return handler
+            return '${BATCH_TMPDIR}' + shq(r._get_path(''))
 
-    def _interpolate_command(self, command, allow_python_results=False):
-        handler = self._interpolate_command_handler(command, allow_python_results)
-        return self._process_command(command, handler)
+        regexes = [_resource.ResourceFile._regex_pattern,
+                   _resource.ResourceGroup._regex_pattern,
+                   _resource.PythonResult._regex_pattern,
+                   Job._regex_pattern,
+                   batch.Batch._regex_pattern]
 
-    def _subst_resource_path_handler(self, local_tmpdir):
-        def handler(match_obj):
-            groups = match_obj.groupdict()
-            assert groups['RESOURCE_FILE'] or groups['RESOURCE_GROUP'] or groups['PYTHON_RESULT']
-            r_uid = match_obj.group()
-            r = self._batch._resource_map.get(r_uid)
-            return shq(r._get_path(local_tmpdir))
-        return handler
+        subst_command = re.sub('(' + ')|('.join(regexes) + ')',
+                               handler,
+                               command)
 
-    def _subst_resource_path(self, command, local_tmpdir):
-        handler = self._subst_resource_path_handler(local_tmpdir)
-        return self._process_command(command, handler)
+        return subst_command
```

Comment:
The block seems to be the same as `self._process_command(handler, command)`. Do we still need `_process_command`?